Extensions/AboutUsParser.js

```diff
@@ -0,0 +1,267 @@
+ */
+function parseString(string) {
+    try {
+        let sections = string.split(C.separator); // We split the string by our main separator into individual sections
+        let parsedSections = [];
+
+        sections.map(section => { // We iterate through every parsed section
+            let para = section.split('\r\n').filter(sentence => sentence.trim() ? true : false); // Splitting sections into paragraphs by new line and removing empty lines
+            //console.log(para.slice(2, para.length));
+            parsedSections.push({ title: para[0], short: para[1], long: para.slice(2,para.length) }); // Pushing a parsed section object into an array
+        });
+
+        return parsedSections; // return array of parsed sections as a callback
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+/**
+ * 
+ * @param {Array} array 
+ */
+function getStringFromArray(array) {
+    let str_builder = '';
+    //console.log(array);
+    for (var par of array)
+        str_builder += new Paragraph(par).getDefaultHtmlCode();
+    
+    return str_builder;
+}
+
+/**
+ * @param {String} Fpath Path where to create the component
+ * @param {String} name File name of the component
+ * @param {String} string String that will be written inside the component
+ * @param {String} language
+ * @param {Boolean} append_file Whether to append or overwrite already existing file 
+ */
+async function createComponent(Fpath = null, name, string, language, append_file = true, fileName = 'school') {
+    Fpath = Fpath ? Fpath : C.path;
+    const component = `:^) ${name} :::`;
+    let path = Path.join(Fpath, language, `${fileName}.html`);
+    writeFile(path, `${component}\r\n${string}`, append_file);
+    console.log(`Successfully created component ${component},\nAppend File Mode: ${append_file}`);
+}
+
+/** Asynchronously writes a string to a file
+ * 
+ * @param {String} path Path of the file
+ * @param {String} string String to write into the file
+ * @param {Boolean} appendFile 
+ * @returns {void}
+ */
+function writeFile(path, string, appendFile) {
+    try {
+        switch (appendFile) {
+            case true:
+                fs.appendFile(path, `${string}\r\n`, { flag: 'a+', mode: 0666 }, (err) => {
+                    if (err) throw new Error(err);
+                });
+                break;
+            case false:
+                fs.writeFile(path, `${string}\r\n`, { flag: 'w+', mode: 0666 }, (err) => {
+                    if (err) throw new Error(err);
+                });
+                break;
+            default:
+                throw new Error('No valid action was chosen for writing a file!');
+        }
+    } catch (error) {
+        console.error(error)
+        return;
+    }
+}
+
+/** Creates a Json file with our data
+ * 
+ * @param {Array} array Array of paragraph objects
+ */
+function saveAsJson(array) {
+    fs.writeFileSync(C.dPath + C.jsonFile, JSON.stringify(array, null, 4), {flag: 'w+', mode: 0666});
```

Comment:
I'm not sure what you either mean by that or I don't see any usage of having it returned to the caller since I don't do anything with later when it's executed.